tests/kernel/timer/timer_api/src/main.c

```diff
@@ -57,7 +57,7 @@ static void init_timer_data(void)
 static void duration_expire(struct k_timer *timer)
 {
 	/** TESTPOINT: expire function */
-	s64_t interval = k_uptime_delta(&tdata.timestamp);
+	volatile s64_t interval = k_uptime_delta(&tdata.timestamp);
```

Comment:
The scope of this variable ends at line 72 right? Or am I missing something?
It should be totally fine if the compile does optimization here.